ci-operator/step-registry/upi/install/heterogeneous/powervs/upi-install-heterogeneous-powervs-commands.sh

```diff
@@ -0,0 +1,203 @@
+          && export PATH=$PATH:/tmp/go/bin && export GOCACHE=/tmp/go && export GOPATH=/tmp/go && export GOMODCACHE="/tmp/go/pkg/mod"
+        if [ -z "$(command -v go)" ]; then
+          echo "Installed go successfully"
+        fi
+      fi
+
+      # build terraform from source v1.5.5
+      cd ${IBMCLOUD_HOME_FOLDER} && curl -L https://github.com/hashicorp/terraform/archive/refs/tags/v1.5.5.tar.gz -o ${IBMCLOUD_HOME_FOLDER}/terraform.tar.gz \
+        && tar -xzf ${IBMCLOUD_HOME_FOLDER}/terraform.tar.gz && cd ${IBMCLOUD_HOME_FOLDER}/terraform-1.5.5 \
+        && go build -ldflags "-w -s -X 'github.com/hashicorp/terraform/version.dev=no'" -o bin/ . && cp bin/terraform /tmp/terraform
+      export PATH=$PATH:/tmp
+      t_ver1=$(/tmp/terraform -version)
+      echo "terraform version: ${t_ver1}"
+
+      export PATH
+      ic version
+      echo "Logging into IBMCLOUD"
+      RESOURCE_GROUP=$(yq -r '.platform.ibmcloud.resourceGroupName' ${SHARED_DIR}/install-config.yaml)
+
+      ic login --apikey @${CLUSTER_PROFILE_DIR}/ibmcloud-api-key -r ${REGION} -g ${RESOURCE_GROUP}
+      ic plugin install -f cloud-internet-services vpc-infrastructure cloud-object-storage power-iaas is
+
+      # create workspace for powervs from cli
+      echo "Display all the variable values:"
+      POWERVS_REGION=$(
+          case "$REGION" in
+              ("jp-osa") echo "osa21" ;;
+              ("eu-gb") echo "lon04" ;;
+              ("ca-tor") echo "mon01" ;;
+              ("br-sao") echo "sao01" ;;
+              ("au-syd") echo "syd04" ;;
+              ("jp-tok") echo "tok04" ;;
+              (*) echo "$REGION" ;;
+          esac)
+      echo "Region is ${REGION} PowerVS region is ${POWERVS_REGION} Resource Group is ${RESOURCE_GROUP}"
+      SERVICE_INSTANCE_OUTPUT=$(ic resource service-instance-create "${WORKSPACE_NAME}" "${SERVICE_NAME}" "${SERVICE_PLAN_NAME}" "${POWERVS_REGION}" -g "${RESOURCE_GROUP}")
+
+      SERVICE_INSTANCE_ID=$(echo "$SERVICE_INSTANCE_OUTPUT" | grep -oE 'GUID:[[:space:]]+[^:[:space:]]+' | awk '{print $2}')
+
+      # Copy the service id to shared_dir to use in deprovisioning step
+      echo ${SERVICE_INSTANCE_ID} > ${SHARED_DIR}/POWERVS_SERVICE_INSTANCE_ID
+      cat ${SHARED_DIR}/POWERVS_SERVICE_INSTANCE_ID
+
+      echo ${RESOURCE_GROUP} > ${SHARED_DIR}/RESOURCE_GROUP
+      echo ${OCP_VERSION} > ${SHARED_DIR}/OCP_VERSION
+
+      # After the workspace is created, invoke the automation code
+      cd ${IBMCLOUD_HOME_FOLDER} \
+        && curl -L https://github.com/IBM/ocp4-upi-compute-powervs/archive/refs/heads/release-${OCP_VERSION}.tar.gz -o ${IBMCLOUD_HOME_FOLDER}/ocp-${OCP_VERSION}.tar.gz \
+        && tar -xzf ${IBMCLOUD_HOME_FOLDER}/ocp-${OCP_VERSION}.tar.gz \
+        && mv ${IBMCLOUD_HOME_FOLDER}/ocp4-upi-compute-powervs-release-${OCP_VERSION} ${IBMCLOUD_HOME_FOLDER}/ocp4-upi-compute-powervs
+
+      echo "Creating the Centos Stream Image"
```

Comment:
What are we actually using this stream image for?

Reply:
Chatting on Slack, summary added to code:

# The CentOS-Stream-8 image is stock-image on PowerVS.
      # This image is available across all PowerVS workspaces.
      # The VMs created using this image are used in support of ignition on PowerVS.

Reply:
In my next commit